### PR TITLE
3804 security vulnerabilities in rest files endpoint

### PR DESCRIPTION
**Pull request by dtaimanov**

### CODE REVIEW

File: jmix-localfs/localfs/src/main/java/io/jmix/localfs/LocalFileStorageProperties.java [hunk 1]
```diff
@@ -37,4 +45,11 @@ public LocalFileStorageProperties(
     public String getStorageDir() {
         return storageDir;
     }
+
+    /**
+     * @see #disablePathCheck
+     */
+    public boolean isDisablePathCheck() {
```
Comment (knstvk) [inline jmix-localfs/localfs/src/main/java/io/jmix/localfs/LocalFileStorageProperties.java:52]: Why do we need this property?
Let's disable any attempt to request a file outside FileStorage root.

File: jmix-rest/rest/src/main/java/io/jmix/rest/RestProperties.java [hunk 2]
```diff
@@ -63,6 +71,13 @@ public boolean isResponseFetchPlanEnabled() {
         return responseFetchPlanEnabled;
     }
 
+    /**
+     * @see #viewFileExtensions
+     */
+    public Set<String> getViewFileExtensions() {
```
Comment (knstvk) [inline jmix-rest/rest/src/main/java/io/jmix/rest/RestProperties.java:77]: I would call it `getInlineEnabledFileExtensions`

File: jmix-core/core/src/main/java/io/jmix/core/impl/FileTransferServiceImpl.java [hunk 3]
```diff
@@ -202,6 +204,12 @@ public ResponseEntity<FileInfoResponse> fileUpload(String name,
             } catch (NumberFormatException ignored) {
             }
 
+            if (size > coreProperties.getMaxFsFileSize().toBytes()) {
```
Comment (knstvk) [inline jmix-core/core/src/main/java/io/jmix/core/impl/FileTransferServiceImpl.java:207]: Checking Content-Length header is not reliable and cannot prevent a deliberate attack. So I'm afraid the whole idea with CoreProperties.maxFsFileSize is useless.

File: jmix-rest/rest/src/main/java/io/jmix/rest/RestProperties.java [hunk 4]
```diff
@@ -38,15 +39,22 @@ public class RestProperties {
     private final int defaultMaxFetchSize;
     private final Map<String, Integer> entityMaxFetchSize;
 
+    /**
+     * File extensions that can be opened for viewing in a browser.
```
Comment (knstvk) [inline jmix-rest/rest/src/main/java/io/jmix/rest/RestProperties.java:43]: Add more details:
"File extensions that can be opened for viewing in a browser by replying with 'Content-Disposition=inline' header"

**Review by knstvk — CHANGES_REQUESTED** [review]

File: jmix-rest/rest/src/main/java/io/jmix/rest/RestProperties.java (hunk 4 above)
Comment (Gavrilov-Ivan) [reply jmix-rest/rest/src/main/java/io/jmix/rest/RestProperties.java]: Done

File: jmix-core/core/src/main/java/io/jmix/core/impl/FileTransferServiceImpl.java (hunk 3 above)
Comment (dtaimanov) [reply jmix-core/core/src/main/java/io/jmix/core/impl/FileTransferServiceImpl.java]: Removed, max file size moved to  LocalFileStorageProperties

File: jmix-core/core/src/main/java/io/jmix/core/CoreProperties.java [hunk 5]
```diff
@@ -106,6 +106,9 @@ public class CoreProperties {
      */
     boolean skipNullOrEmptyConditionsByDefault;
 
+
```
Comment (Gavrilov-Ivan) [inline jmix-core/core/src/main/java/io/jmix/core/CoreProperties.java:109]: Remove empty lines - no actual changes in this file

File: jmix-core/core/src/main/java/io/jmix/core/impl/FileTransferServiceImpl.java [hunk 6]
```diff
@@ -51,6 +51,8 @@ public class FileTransferServiceImpl implements FileTransferService {
 
     @Autowired
     private FileStorageLocator fileStorageLocator;
+    @Autowired
+    private CoreProperties coreProperties;
```
Comment (Gavrilov-Ivan) [inline jmix-core/core/src/main/java/io/jmix/core/impl/FileTransferServiceImpl.java:55]: No more usage - remove field.
No actual changes in this file (import refactoring is not a part of the issue and should not present here).

File: jmix-core/core/src/test/java/test_support/TestCoreProperties.java [hunk 7]
```diff
@@ -76,6 +77,7 @@ public static class Builder {
         Duration triggerFilesProcessInterval = Duration.ofSeconds(5000);
         boolean roundDecimalValueByFormat = true;
         boolean skipNullOrEmptyConditionsByDefault = false;
+        DataSize maxFsFileSize = DataSize.ofMegabytes(100);
```
Comment (Gavrilov-Ivan) [inline jmix-core/core/src/test/java/test_support/TestCoreProperties.java:80]: Do we still need this here (after moving to LocalFS properties)?

**Review by Gavrilov-Ivan — CHANGES_REQUESTED** [review]

**Review by Gavrilov-Ivan — APPROVED** [review]

**Pull request merged**